ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/AuditPasswordAuthenticator.java

```diff
@@ -88,10 +87,10 @@ public void setup()
     }
 
     @Override
-    public SaslNegotiator newSaslNegotiator(InetAddress clientAddress)
+    public SaslNegotiator newSaslNegotiator()
     {
-        LOG.debug("Setting up SASL negotiation with {}", clientAddress);
-        return new AuditPlainTextSaslAuthenticator(clientAddress, wrappedAuthenticator.newSaslNegotiator(clientAddress));
+        LOG.debug("Setting up SASL negotiation with client peer");
+        return new AuditPlainTextSaslAuthenticator(wrappedAuthenticator.newSaslNegotiator());
```

Comment:
We should consider to do some research on whether it is possible to extract client address some how without being to hacky.

Reply:
Well, this method is called by the ServerConnection class, which has access to the ClientState which in turn has the address... 
BUT there is no way in Java (as far as I know) to get access to the calling instance from somewhere in the code. Even if we know which class that is calling us, we still need the instance to be able to use reflection to access that instance member data. (same goes for the AuditPlainTextSaslAuthenticator#getAuthenticatedUser() method below)
So I don't think it can be done...

https://stackoverflow.com/questions/43072542/can-i-get-the-instance-of-the-calling-object-in-java
Basically says:
"Short answer: you can't.
Long answer: you can get it if you either run it on a debugger, or your own custom JVM."